packages/@aws-cdk/aws-cloudformation-codepipeline/lib/pipeline-action.ts

```diff
@@ -0,0 +1,231 @@
+     */
+    capabilities?: CloudFormationCapabilities[];
+
+    /**
+     * A name for the output file, such as CreateStackOutput.json. AWS CodePipeline adds the file to
+     * the output artifact after performing the specified action.
+     */
+    outputFileName?: string;
+
+    /**
+     * A JSON object that specifies values for template parameters. If you specify parameters that
+     * are also specified in the template configuration file, these values override them. All
+     * parameter names must be present in the stack template.
+     *
+     * Note: There is a maximum size limit of 1 kilobyte for the JSON object that can be stored in
+     * the ParameterOverrides property.
+     *
+     * We recommend that you use the template configuration file to specify most of your parameter
+     * values. Use parameter overrides to specify only dynamic parameter values (values that are
+     * unknown until you run the pipeline).
+     */
+    parameterOverrides?: { [name: string]: any };
+
+    /**
+     * The template configuration file can contain template parameter values and a stack policy.
+     * Note that if you include sensitive information, such as passwords, restrict access to this
+     * file. For more information, see AWS CloudFormation Artifacts.
+     */
+    templateConfiguration?: ArtifactPath;
+}
+
+export abstract class CloudFormationAction extends DeployAction {
+    constructor(parent: Stage, name: string, configuration?: any) {
+        super(parent, name, 'CloudFormation',  { minInputs: 0, maxInputs: 10, minOutputs: 0, maxOutputs: 1 }, configuration);
+    }
+}
+
+export interface ExecuteChangeSetOptions extends CloudFormationCommonOptions {
+
+    /**
+     * The stack name to execute a change set against.
+     */
+    stackName: string;
+
+    /**
+     * The name of an existing change set or a new change set that you want to create for the
+     * specified stack.
+     */
+    changeSetName: string;
+}
+
+/**
+ * Executes a change set.
+ */
+export class ExecuteChangeSet extends CloudFormationAction {
+    constructor(parent: Stage, name: string, options: ExecuteChangeSetOptions) {
+        super(parent, name, {
+            ActionMode: 'CHANGE_SET_EXECUTE',
+            Capabilities: options.capabilities,
+            ChangeSetName: options.changeSetName,
+            OutputFileName: options.outputFileName,
+        });
+    }
+}
+
+export interface CreateReplaceChangeSetOptions extends CloudFormationCommonOptions {
+
+    /**
+     * The stack name to create/replace a change set for.
+     */
+    stackName: string;
+
+    /**
+     * The name of an existing change set or a new change set that you want to create for the
+     * specified stack.
+     */
+    changeSetName: string;
+
+    /**
+     * The arn of the IAM service role that AWS CloudFormation assumes
+     * when it operates on resources in the specified stack.
+     */
+    roleArn: RoleArn;
```

Comment:
This should be `Role` and optional (default should be to auto-create a role by the construct)

Reply:
Tricky. Normally when we follow this pattern, we autocreate a `Role` with the right permissions. 

In this case though, the "right permissions" are `*:*`, because there's no telling what the CloudFormation might do. Except... autocreating a role that has `*:*` feels risky. So the better solution might be to punt it to the consumer, and have them knowingly opt in.

In the same vein, I wonder if we should default the IAM capabilities to... you know... work. Everyone's going to want this and hence specify the IAM capabilities. Should we make people opt out to improve convenience in the 99% case?

Reply:
We could still create a role with no statements and require that the user explicitly call `addToRolePolicy` with whatever permissions they want.

As sugar, we can implement a props flag such as `adminRights: boolean` which will auto-add the "*/*" and set the IAM capabilities to all